pam-u2f.c

```diff
@@ -305,6 +349,13 @@ int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
     buf = NULL;
   }
 
+  if (cfg->authpending_file) {
+    if (should_free_authpending_file) {
```

Comment:
Isn't it enough to check should_free?
Also if any of the previous calls fails should_free is uninitialized here. It should be declared a bit higher up